src/main/scala/firrtl/Utils.scala

```diff
@@ -188,6 +188,17 @@ object Utils extends LazyLogging {
     case sx => sx
   }
 
+  /** Returns true if PrimOp is a Not, false otherwise */
+  def isNot(op: PrimOp): Boolean = op match {
+    case Not => true
+    case _ => false
+  }
+  /** Returns true if Expression is a Not PrimOp, false otherwise */
+  def isNot(expr: Expression): Boolean = expr match {
+    case DoPrim(op, _,_,_) if isNot(op) => true
```

Comment:
Related to the above
```suggestion
    case DoPrim(Not, _,_,_) => true
```